src/main/java/org/eclipse/emfcloud/jackson/databind/deser/EDataTypeDeserializer.java

```diff
@@ -26,6 +25,11 @@
 
 public class EDataTypeDeserializer extends JsonDeserializer<Object> {
 
+   public static boolean isJavaLangType(final EDataType dataType) {
+      String instanceClassName = dataType.getInstanceClassName();
+      return instanceClassName.startsWith("java.lang.") || instanceClassName.indexOf('.') < 0;
```

Comment:
When you define a data type, it is your duty (contract to be fulfilled) to implement the create/convert methods if you need to support serialisation. That's part of the job when using EMF. Hence, IMO (as detailed below) we should be even more greedy. I'm not sure why the original implementation doesn't rely on EFactory in general, I'd assume that all pre-defined data types in Ecore support serialization and just should work. However, the tests would need to be re-written, since there are assumptions there of the specific format that may not match Ecore's. Some cases won't work, e.g. Object[] is handled not by a general serialization of Objects, but using JSON-specific logic, if I understand it correctly. I would start by reverting the logic, use EFactory in all cases where the data type has the serializable flag set and then see if there are special cases that could need special handling.

Reply:
If we want to make certain data types easier to support, we can also define a separate Emfjson model with pre-defined data types, so anyone can import it and use its data types. This can be cleaner than hard-coding the logic in the serialization class.
An orthogonal option is to introduce some annotations on the data type, for controlling how emfjson-specific logic should be used for serialization.

Reply:
We are also not sure why the original implementation did not rely on EFactory in general but I think it is the right way forward. However, as to not break backwards compatibility, we should use an EAnnotation to allow switching between strategies as there might be some customizations we are not aware of. There is already little support for annotations in `JsonAnnotations`.